### PR TITLE
Add compat data for CanvasGradient

### PR DESCRIPTION
**Pull request by maboa**
I made the following assumptions:

1. Used `true` when a value of `"(Yes)"` was encountered (ie https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient/addColorStop)

2. Replaced `"1.0"` with `null` for Firefox Android as according to the linter Firefox Android versions begin at version 4.

3. Omitted `(1.9.2)` from the version number for Firefox(Gecko) `3.6 (1.9.2)` column as Gecko is not mentioned in the new JSON. (https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient)

Happy to amend :)

### CODE REVIEW

**Comment by teoli2003** [comment]
@maboa Please remove "ie_mobile" as we removed it a few minutes ago. (No worry, this kind of change happens about once a Quarter)

**Comment by Elchi3** [comment]
>  Used true when a value of "(Yes)" was encountered (ie https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient/addColorStop)

Yes, this is exactly the idea behind true values.

>  Replaced "1.0" with null for Firefox Android as according to the linter Firefox Android versions begin at version 4.

You can use version 4 instead of null here. Everything <4 for Fx for Android is actually 4.

>  Omitted (1.9.2) from the version number for Firefox(Gecko) 3.6 (1.9.2) column as Gecko is not mentioned in the new JSON. (https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient)

Yes, our goal is to no longer talk about Gecko on MDN generally. Please use the Firefox versions only.

**Comment by maboa** [comment]
Thanks for the feedback both. Done and done :)

File: api/CanvasGradient.json [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+          },
+          "edge": {
+            "version_added": true
+          },
+          "edge_mobile": {
+            "version_added": true
+          },
+          "firefox": {
+            "version_added": "3.6",
+            "notes": "Before Gecko 5.0, (Firefox 5.0 / Thunderbird 5.0 / SeaMonkey 2.2), specifying non-finite values when adding color stops through a call to <code>addColorStop()</code> incorrectly throws <code>SYNTAX_ERR</code> instead of <code>INDEX_SIZE_ERR</code>."
```
Comment (Elchi3) [inline api/CanvasGradient.json]: This note should just read "Before Firefox 5, specifying non-finite (...)".
We're not mentioning Gecko, Thunderbird and SeaMonkey anymore.

Comment (maboa) [reply api/CanvasGradient.json]: OK

**Pull request merged**